wrapper/src/main/java/software/amazon/jdbc/util/PropertyUtils.java

```diff
@@ -116,72 +116,4 @@ public static void setPropertyOnTarget(
     }
     return copy;
   }
-
-  @SuppressWarnings("checkstyle:LocalVariableName")
-  public static @Nullable Properties parseProperties(String url, @Nullable Properties defaults) {
-
-    String urlServer = url;
-    String urlArgs = "";
-
-    int qPos = url.indexOf('?');
-    if (qPos != -1) {
-      urlServer = url.substring(0, qPos);
-      urlArgs = url.substring(qPos + 1);
-    }
-
-    Properties propertiesFromUrl = new Properties();
-
-    // TODO: allow user to disable database parsing
-    int protocolPos = urlServer.indexOf("//");
-    if (protocolPos != -1) {
-      protocolPos += 2;
-    }
-    int dPos = urlServer.indexOf("/", protocolPos);
-    if (dPos != -1) {
-      String database = urlServer.substring(dPos + 1);
-      if (!database.isEmpty()) {
-        PropertyDefinition.DATABASE.set(propertiesFromUrl, database);
-      }
-    }
-
-    String[] args = urlArgs.split("&");
-
-    for (String token : args) {
-
-      if (token.isEmpty()) {
-        continue;
-      }
-      int pos = token.indexOf('=');
-      if (pos == -1) {
-        propertiesFromUrl.setProperty(token, "");
-      } else {
-        String pName = token.substring(0, pos);
-        String pValue = urlDecode(token.substring(pos + 1));
-        if (pValue == null) {
-          return null;
-        }
-        propertiesFromUrl.setProperty(pName, pValue);
-      }
-    }
-
-    Properties result = new Properties();
-    result.putAll(propertiesFromUrl);
-    if (defaults != null) {
-      defaults.forEach(result::putIfAbsent);
-    }
-
-    return result;
-  }
-
-  private static @Nullable String urlDecode(String url) {
-    try {
-      return StringUtils.decode(url);
```

Comment:
I can't find any place in ConnectionUrlParser class where it uses decode(). Can ConnectionUrlParser handle encoded parameters properly?

Another point to check if ConnectionUrlParser can properly handle parameters with no value? 
"?paramWithNoValue&otherParam=otherParamValue"

Reply:
Put the url decode method into ConnectionUrlParser, added unit tests testing some encoded parameter values.